wa/framework/uiauto-androidx/gradle.properties

```diff
@@ -0,0 +1 @@
+android.useAndroidX=true
```

Comment:
Apologies for the late comment on this, but IIUC it seems that this is currently the main meaningful difference between the `uiauto` and `uiauto-androidx` directory trees?

I.e. If I do a diff on these these directories, this is my output:
```
Common subdirectories: uiauto/app and uiauto-androidx/app
diff --color uiauto/build.gradle uiauto-androidx/build.gradle
5d4
<         jcenter()
6a6
>         mavenCentral()
9c9
<         classpath 'com.android.tools.build:gradle:7.2.1'
---
>         classpath 'com.android.tools.build:gradle:8.5.0'
11d10
<
13c12
<         // in the individual module build.gradle files
---
>     // in the individual module build.gradle files
19d17
<         jcenter()
20a19
>         mavenCentral()
diff --color uiauto/build.sh uiauto-androidx/build.sh
2c2
< #    Copyright 2013-2017 ARM Limited
---
> #    Copyright 2013-2024 ARM Limited
Common subdirectories: uiauto/gradle and uiauto-androidx/gradle
Only in uiauto-androidx: gradle.properties
Binary files uiauto/uiauto.aar and uiauto-androidx/uiauto.aar differ
```

Therefore I'm wondering if we really need the two trees here currently as I'm concerned about the duplication while we transition over to the new libraries.

Do you think it would be possible to instead just have 2 gradle.properties files and 2 build scripts in the same tree?
One that sets this property to True and one that sets it to False? That way we could maintain the same source tree but would still be able to build both versions of the library? 

Plmk if I'm missing something that would make this a non-starter.

Reply:
On my end, the differences between wa/framework/uiauto and wa/framework/uiauto-androidx are a bit more significant:

`diff -up -r uiauto/app/build.gradle uiauto-androidx/app/build.gradle
--- uiauto/app/build.gradle	2024-07-26 15:42:00.889125106 +0100
+++ uiauto-androidx/app/build.gradle	2024-07-26 15:40:54.948614211 +0100
@@ -1,18 +1,18 @@
 apply plugin: 'com.android.library'
 
 android {
+    namespace "com.arm.wa.uiauto"
     compileSdkVersion 28
-    buildToolsVersion '28.0.3'
     defaultConfig {
         minSdkVersion 18
         targetSdkVersion 28
-        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
 }
 
 dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
-    implementation 'com.android.support.test:runner:0.5'
-    implementation 'com.android.support.test:rules:0.5'
-    implementation 'com.android.support.test.uiautomator:uiautomator-v18:2.1.2'
+    implementation 'androidx.test:runner:1.6.1'
+    implementation 'androidx.test:rules:1.6.1'
+    implementation 'androidx.test.uiautomator:uiautomator-v18:2.2.0-alpha1'
 }
diff -up -r uiauto/app/src/main/AndroidManifest.xml uiauto-androidx/app/src/main/AndroidManifest.xml
--- uiauto/app/src/main/AndroidManifest.xml	2024-07-26 15:40:54.948614211 +0100
+++ uiauto-androidx/app/src/main/AndroidManifest.xml	2024-07-26 15:40:54.948614211 +0100
@@ -1,5 +1,4 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.arm.wa.uiauto">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
     <uses-permission android:name="android.permission.READ_LOGS"/>
 
diff -up -r uiauto/app/src/main/java/com/arm/wa/uiauto/ActionLogger.java uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/ActionLogger.java
--- uiauto/app/src/main/java/com/arm/wa/uiauto/ActionLogger.java	2024-07-26 15:40:54.948614211 +0100
+++ uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/ActionLogger.java	2024-08-29 11:09:59.184231510 +0100
@@ -1,4 +1,4 @@
-/*    Copyright 2014-2018 ARM Limited
+/*    Copyright 2014-2024 ARM Limited
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
diff -up -r uiauto/app/src/main/java/com/arm/wa/uiauto/ApplaunchInterface.java uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/ApplaunchInterface.java
--- uiauto/app/src/main/java/com/arm/wa/uiauto/ApplaunchInterface.java	2024-07-26 15:40:54.948614211 +0100
+++ uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/ApplaunchInterface.java	2024-08-29 11:09:59.184231510 +0100
@@ -1,4 +1,4 @@
-/*    Copyright 2013-2017 ARM Limited
+/*    Copyright 2013-2024 ARM Limited
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -16,7 +16,7 @@
 package com.arm.wa.uiauto;
 
 import android.os.Bundle;
-import android.support.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObject;
 
 /**
  * ApplaunchInterface.java
diff -up -r uiauto/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java
--- uiauto/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java	2024-07-26 15:40:54.952614242 +0100
+++ uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/BaseUiAutomation.java	2024-08-29 11:09:59.188231543 +0100
@@ -1,4 +1,4 @@
-/*    Copyright 2013-2018 ARM Limited
+/*    Copyright 2013-2024 ARM Limited
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -22,13 +22,13 @@ import android.content.Context;
 import android.graphics.Point;
 import android.graphics.Rect;
 
-import android.support.test.InstrumentationRegistry;
-import android.support.test.uiautomator.UiDevice;
-import android.support.test.uiautomator.UiObject;
-import android.support.test.uiautomator.UiObjectNotFoundException;
-import android.support.test.uiautomator.UiSelector;
-import android.support.test.uiautomator.UiWatcher;
-import android.support.test.uiautomator.UiScrollable;
+import androidx.test.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+import androidx.test.uiautomator.UiWatcher;
+import androidx.test.uiautomator.UiScrollable;
 
 import org.junit.Before;
 import org.junit.Test;
@@ -39,7 +39,7 @@ import java.io.InputStreamReader;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
-import static android.support.test.InstrumentationRegistry.getArguments;
+import static androidx.test.InstrumentationRegistry.getArguments;
 
 public class BaseUiAutomation {
 
diff -up -r uiauto/app/src/main/java/com/arm/wa/uiauto/UiAutoUtils.java uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/UiAutoUtils.java
--- uiauto/app/src/main/java/com/arm/wa/uiauto/UiAutoUtils.java	2024-07-26 15:40:54.952614242 +0100
+++ uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/UiAutoUtils.java	2024-08-29 11:09:59.188231543 +0100
@@ -1,4 +1,4 @@
-/*    Copyright 2013-2017 ARM Limited
+/*    Copyright 2013-2024 ARM Limited
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
diff -up -r uiauto/app/src/main/java/com/arm/wa/uiauto/UxPerfUiAutomation.java uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/UxPerfUiAutomation.java
--- uiauto/app/src/main/java/com/arm/wa/uiauto/UxPerfUiAutomation.java	2024-07-26 15:40:54.952614242 +0100
+++ uiauto-androidx/app/src/main/java/com/arm/wa/uiauto/UxPerfUiAutomation.java	2024-08-29 11:09:59.188231543 +0100
@@ -1,4 +1,4 @@
-/*    Copyright 2013-2017 ARM Limited
+/*    Copyright 2013-2024 ARM Limited
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
diff -up -r uiauto/build.gradle uiauto-androidx/build.gradle
--- uiauto/build.gradle	2024-07-26 15:41:53.129065105 +0100
+++ uiauto-androidx/build.gradle	2024-07-26 15:40:54.948614211 +0100
@@ -2,22 +2,21 @@
 
 buildscript {
     repositories {
-        jcenter()
         google()
+        mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.2.1'
-
+        classpath 'com.android.tools.build:gradle:8.5.0'
 
     // NOTE: Do not place your application dependencies here; they belong
-        // in the individual module build.gradle files
+    // in the individual module build.gradle files
     }
 }
 
 allprojects {
     repositories {
-        jcenter()
         google()
+        mavenCentral()
     }
 }
 
diff -up -r uiauto/build.sh uiauto-androidx/build.sh
--- uiauto/build.sh	2024-07-26 15:40:54.952614242 +0100
+++ uiauto-androidx/build.sh	2024-08-29 11:09:59.188231543 +0100
@@ -1,5 +1,5 @@
 #!/bin/bash
-#    Copyright 2013-2017 ARM Limited
+#    Copyright 2013-2024 ARM Limited
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
diff -up -r uiauto/gradle/wrapper/gradle-wrapper.properties uiauto-androidx/gradle/wrapper/gradle-wrapper.properties
--- uiauto/gradle/wrapper/gradle-wrapper.properties	2024-07-26 15:40:54.952614242 +0100
+++ uiauto-androidx/gradle/wrapper/gradle-wrapper.properties	2024-08-29 11:09:59.188231543 +0100
@@ -1,6 +1,6 @@
-#Wed May 03 15:42:44 BST 2017
+#Tue 27 Aug 2024 12:28:27 PM BST
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-all.zip
Only in uiauto-androidx: gradle.properties`

For the androidx version there were version bumps for the libraries, removal of deprecated properties (buildToolsVersion IIRC), updates to the JUNIT runner (to use the androidx one), the gradle version, the repositories to fetch libraries from and changes to the uiauto app itself to use androidx libraries.

The way I see it, we want to update all the other apps to use the updated uiauto framework using androidx, because the older android support library is now deprecated. Unfortunately I was never able to chase the right versions of the apk's for some of the workloads. So it is a bit tricky to test and make sure they work. In the end, I considered keeping the original uiauto framework as-is for now, in the hopes it would still work for whoever still has those valid apk's.

I'm not sure I have the cycles to look into this at the moment, but we could probably coordinate internally to get this updated so we don't have to use deprecated libraries.